core/src/main/java/org/apache/accumulo/core/fate/accumulo/AccumuloStore.java

```diff
@@ -70,7 +70,8 @@ public AccumuloStore(ClientContext context) {
   public long create() {
     long tid = RANDOM.get().nextLong() & 0x7fffffffffffffffL;
 
-    newMutator(tid).putStatus(TStatus.NEW).putCreateTime(System.currentTimeMillis()).mutate();
+    newMutator(tid).requireStatus().putStatus(TStatus.NEW).putCreateTime(System.currentTimeMillis())
```

Comment:
We need to retry when this fails because the conditiona was not accepted.  Not sure of the best way to do this.  Could have a special exception and catch that OR make mutate() return something.  Also need to test this if possible.

Maybe we could make the next id a function that can be overridden by a test class.

```java
  // exist to be overridden by test so can test trying to create existing id
 protected long getNextId() {
    return RANDOM.get().nextLong() & 0x7fffffffffffffffL;
}

public long create() {
   long tid = getNextId();

   while(true) {
      // attempt to persist id
      // if successful return tid
     /// if not successful then try again  : tid = getNextId(); 
      //sleep or use retry obj
    }

```

Feel free to ignore the comments about testing. Not sure about my testing comment, maybe testing needs to be at a lower level like is already being done. Even w/o the changes for testing, do need to retry somehow.

Reply:
This is a good idea. I was also thinking about something similar: how we can't tell which condition attached to a mutation has caused it to fail. It would be nice to have something like an optional message that could be added to the ConditionalWriter.Result object so that when the result is returned, we could know what caused the mutation to fail.

As things are now though, I am not sure how we could trigger a retry, since there could be several things that cause an exception to be thrown, we don't know if its this case where we want a retry to happen.

Reply:
I experimented with making mutate() return something, but that seemed a bit dangerous as it would be easy to ignore the return value.  Realized maybe adding a new `tryMutate()` method might be good because the prefix `try` implies a return value, this is following some of the methods in the java api that have a `try` prefix.  Below is the commit where I experimented with this.

https://github.com/keith-turner/accumulo/commit/f70f6bc65a879515bc803358d5219cafbb90ec57

It also adds a new enum because we only want to handle a subset of the statuses that conditional writer can return.

Reply:
Also I was curious what the potential for a collision was and found : 

https://en.wikipedia.org/wiki/Birthday_attack

which has a nice table.  If I am reading the table correctly, for a 64 bit integer if there are 6.1M entries in the fate table then we would have a 1 in 1M chance of a collision.

Reply:
Added this in 9240cad